### ISSUE TITLE
Why do you not use the material-components-web JS?

### ISSUE THREAD

**Issue by davidharting** [kradio3/react-mdc-web#9]
@kradio3, Why do you not use the Google-supported JS files for material-components-web components? The material-components-web repo provides examples of how to integrate their JS into React components. Not criticizing the decision to not use their JS, just wondering what your thought process was.

**Comment by kradio3** [comment]
@davidharting ,this is good question 
Main reasons are library size and performance. Look at the [Snackbar](https://github.com/kradio3/react-mdc-web/blob/master/src/Snackbar/Snackbar.js). It takes 102 lines
Compare it with google's JS + react adapter. 
1. [Snackbar foundation class](https://github.com/material-components/material-components-web/blob/master/packages/mdc-snackbar/foundation.js) - 158 lines
2. And adapter for React (for example [haoxins's repo](https://github.com/haoxins/react-mdc/blob/master/lib/snackbar.js) ) - 158 lines

Performace. In case of using adapters, component has more complex state. You can compare snackbar implementations above. Without adapter it's stateless, with adapter - it's stateful.

**Comment by davidharting** [comment]
Thank you for the responses @kradio3! Makes a lot of sense. The Foundation / Adapter pattern does seem to be very heavy-handed. 

Very excited about this project. Thank you for your great work!

**Comment by davidharting** [comment]
Closing this issue. Got the information I was looking for.

**Comment by piotr-cz** [comment]
As a reference, I'm cross-linking comments on such approach on [mdc issues](https://github.com/material-components/material-components-web/issues/407).
TL;DR it's fine
